File: src/types/node.ts
export interface NodeInfo {
    name: string;
    protocol: string;
    address: string;
    port: number;
    fileName: string;
}

export interface NodeConfig {
    outbounds: OutboundConfig[];
}

export interface OutboundConfig {
    tag: string;
    protocol: string;
    settings?: Record<string, unknown>;
    streamSettings?: Record<string, unknown>;
}

File: src-tauri/src/models/node.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct NodeInfo {
    pub name: String,
    pub protocol: String,
    pub address: String,
    pub port: u16,
    pub file_name: String,
}
